### PR TITLE
Add com.adilhanney.ricochlime

### PR DESCRIPTION
**Pull request by adil192**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. <s>If not, I contacted upstream developers about submitting their software to Flathub.</s> **Link:** https://github.com/adil192/ricochlime
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] <s>Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*</s>

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### To-do

- [x] Use offline fonts and remove internet permission
- [x] Restructure the linux build archive to include the icons and metadata

### CODE REVIEW

File: update.sh
Comment (hfiguiere) [inline update.sh:1]: instead you should use the https://github.com/flathub/flatpak-external-data-checker

File: com.adilhanney.ricochlime.json [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+				"install -Dm644 com.adilhanney.ricochlime.metainfo.xml -t /app/share/metainfo/",
+		
+				"install -Dm644 icon-512x512.png /app/share/icons/hicolor/512x512/apps/com.adilhanney.ricochlime.png",
+	
+				"install -Dm644 com.adilhanney.ricochlime.desktop -t /app/share/applications/"
+			],
+			"sources": [
+				{
+					"type": "file",
+					"url": "https://raw.githubusercontent.com/adil192/ricochlime/v0.4.2/flatpak/com.adilhanney.ricochlime.metainfo.xml", "sha256": "e95ac99d366f1d9ad17bfeb20eb845ad471de126b9d9e66a57282b67ad0058b6"
```
Comment (hfiguiere) [inline com.adilhanney.ricochlime.json:43]:
```suggestion
					"url": "https://raw.githubusercontent.com/adil192/ricochlime/v0.4.2/flatpak/com.adilhanney.ricochlime.metainfo.xml", 
					"sha256": "e95ac99d366f1d9ad17bfeb20eb845ad471de126b9d9e66a57282b67ad0058b6"
```

**Comment by adil192** [comment]
Marking this as a draft since I want to wait until I have access to my linux device

**Comment by adil192** [comment]
bot, build com.adilhanney.ricochlime

**Comment by adil192** [comment]
bot, build com.adilhanney.ricochlime

File: com.adilhanney.ricochlime.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+				"ln -s /app/Ricochlime/ricochlime /app/bin/ricochlime",
+				"ln -s /app/Ricochlime/lib/* /app/lib/",
+				"install -Dm644 com.adilhanney.ricochlime.metainfo.xml -t /app/share/metainfo/",
+				"install -Dm644 icon-512x512.png /app/share/icons/hicolor/512x512/apps/com.adilhanney.ricochlime.png",
+				"install -Dm644 com.adilhanney.ricochlime.desktop -t /app/share/applications/"
+			],
+			"sources": [
+				{
+					"type": "file",
+					"url": "https://raw.githubusercontent.com/adil192/ricochlime/v0.4.2/flatpak/com.adilhanney.ricochlime.metainfo.xml",
```
Comment (hfiguiere) [inline com.adilhanney.ricochlime.json:39]: any reason why these files are not in the tarball?

Comment (adil192) [reply com.adilhanney.ricochlime.json:39]: The `tar.gz` file is just the output of the flutter build command, which I use for both Flatpak and AppImage. https://github.com/flathub/flathub/pull/3459#discussion_r963122325

Comment (hfiguiere) [reply com.adilhanney.ricochlime.json:39]: and they could be used by both the same way. A Linux app without a .desktop or an icon, it is a bug. The appstream file is extra bonus (that flatpak require)

So it should be part of the build artifacts.

File: update.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+flatpak run \
+  --filesystem=$(pwd) \
+  org.flathub.flatpak-external-data-checker \
+  com.adilhanney.ricochlime.json \
+  --update
```
Comment (hfiguiere) [inline update.sh:7]: this will be done automatically on github....

File: com.adilhanney.ricochlime.json [hunk 4]
```diff
@@ -0,0 +1,85 @@
+{
+	"app-id": "com.adilhanney.ricochlime",
+	"runtime": "org.freedesktop.Platform",
+	"runtime-version": "22.08",
+	"sdk": "org.freedesktop.Sdk",
+	"command": "ricochlime",
+	"separate-locales": false,
+	"finish-args": [
+		"--share=network",
```
Comment (hfiguiere) [inline com.adilhanney.ricochlime.json:9]: does it need network?

Comment (adil192) [reply com.adilhanney.ricochlime.json:9]: For now, it does for the [google_fonts](https://pub.dev/packages/google_fonts) package. In the future, I'll include the fonts offline

Comment (hfiguiere) [reply com.adilhanney.ricochlime.json:9]: they should be in the package instead.

File: update.sh (hunk 3 above)
Comment (adil192) [reply update.sh:7]: I just thought it'd be handy to have the command written down somewhere. If you want me to remove it I can do

Comment (hfiguiere) [reply update.sh:7]: it doesn't matter.

**Comment by adil192** [comment]
bot, build com.adilhanney.ricochlime

**Comment by adil192** [comment]
bot, build com.adilhanney.ricochlime

File: com.adilhanney.ricochlime.json [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+				"mkdir -p /app",
+				"tar -xf Ricochlime_*.tar.gz -C /app",
+
+				"mkdir -p /app/bin",
+				"ln -s /app/ricochlime /app/bin/ricochlime"
+			],
+			"sources": [
+				{
+					"type": "file",
+					"url": "https://github.com/adil192/ricochlime/releases/download/v0.5.1/Ricochlime_v0.5.1.tar.gz",
```
[the remark this answers is not in the document]
Comment (adil192) [reply com.adilhanney.ricochlime.json:29]: Sorry that's a bit of an oversight. I need to build another archive for arm64

**Comment by adil192** [comment]
bot, build com.adilhanney.ricochlime

**Comment by hfiguiere** [comment]
> That's what we do for F-Droid, but maybe Flathub doesn't allow internet access while building?

Build MUST be offline.

**Pull request closed without merging**